Taskfile.yml

```diff
@@ -28,23 +28,23 @@ tasks:
 
   test:
     desc: Run python integration tests.
+    deps:
+      - install
     cmds:
-      - python3 -m unittest discover provider/cmd/pulumi-resource-nuage/tests/
+      - pytest --capture=sys provider/cmd/pulumi-resource-nuage/pytests/tests/ -s
 
   install:
     desc: Install requirements for integration tests.
     deps:
       - virtualenv
     cmds:
-      - source provider/cmd/pulumi-resource-nuage/tests/venv/bin/activate
-      - pip install -U pip wheel
-      - pip install -r provider/cmd/pulumi-resource-nuage/requirements.txt
+      - provider/cmd/pulumi-resource-nuage/pytests/venv/bin/pip install -r provider/cmd/pulumi-resource-nuage/pytests/requirements.txt
```

Comment:
You could use the dir parameter on the task definition

Reply:
Also, why not use Poetry ?

At least this way we could easily upgrade dependencies